main.go

```diff
@@ -34,7 +34,7 @@ func main() {
 	}
 
 	stopChan := make(chan string)
-	go master(stopChan)
+	go master(&stopChan)
```

Comment:
The channel doesn't seemed to get recreated if you don't pass the pointer/address with `&`

https://play.golang.org/p/MtbGgktDZK